orquesta/tests/unit/conducting/test_workflow_conductor_rerun.py

```diff
@@ -0,0 +1,425 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+from orquesta import conducting
+from orquesta.specs import native as native_specs
+from orquesta import statuses
+from orquesta.tests.unit import base as test_base
+from orquesta.utils import dictionary as dict_util
+
+
+PARALLEL_WF_DEF = """
+version: 1.0
+
+description: A basic branching workflow.
+
+vars:
+  - var1: abc
+  - var2: 234
+
+tasks:
+  # branch 1
+  task1:
+    action: core.noop
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - var1: 'xyz'
+        do: task2
+  task2:
+    action: core.noop
+
+  # branch 2
+  task3:
+    action: core.noop
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - var2: 123
+        do: task4
+  task4:
+    action: core.noop
+
+output:
+  - var1: <% ctx().var1 %>
+  - var2: <% ctx().var2 %>
+"""
+
+JOIN_WF_DEF = """
+version: 1.0
+
+description: A basic branching workflow.
+
+tasks:
+  # branch 1
+  task1:
+    action: core.noop
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - var1: 'xyz'
+        do: task3
+
+  # branch 2
+  task2:
+    action: core.noop
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - var2: 123
+        do: task3
+
+  # joining branch
+  task3:
+    join: all
+    action: core.noop
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - var3: True
+        do: task4
+  task4:
+    action: core.noop
+"""
+
+
+class WorkflowConductorRerunTest(test_base.WorkflowConductorTest):
+
+    def pre_rerun_workflow(self, wf_def):
+        spec = native_specs.WorkflowSpec(wf_def)
+        self.assertDictEqual(spec.inspect(), {})
+
+        # Run the workflow.
+        conductor = conducting.WorkflowConductor(spec)
+        conductor.request_workflow_status(statuses.RUNNING)
+
+        return conductor
+
+    def validate_request_workflow_rerun(self, conductor, rerun_tasks, number_tasks):
```

Comment:
Can you rename this function to `assert_workflow_rerun_request`?  This is more consistent naming with other test methods that does similar validation.  Thanks.